w3af_batch

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+
+import logging
+import argparse
+import signal
+import threading as th
+import multiprocessing as mp
+import multiprocessing.dummy as mpd
+
+
+WAIT_TIMEOUT = 30  # in seconds
+STOPSCAN = mp.Event()
+logger = logging.getLogger('w3af_batch')
+
+
+def cancel_scan(*args):
+    """
+    Dummy SIGINT hanlder which stops target list processing.
+    """
+    STOPSCAN.set()
+    raise KeyboardInterrupt('Scan is terminated by user request')
+
+
+def execute_scan(target, profile, timeout):
+    """
+    Execute w3af scan. Try to stop it after timeout seconds.
+    """
+    def stop_scan(scanner):
+        """
+        Stop w3af scan after timeout.
+        """
+        if scanner.status.is_running():
+            scanner.stop()
+            logger.info('Stop scan of %s because of configured timeout',
+                        target)
+
+    from w3af.core.controllers.w3afCore import w3afCore
+    from w3af.core.data.kb.knowledge_base import kb
+
+    scanner = w3afCore()
+    scanner.profiles.use_profile(profile)
+    option_list = scanner.target.get_options()
+    option_list['target'].set_value(target)
+    scanner.target.set_options(option_list)
+    scanner.plugins.init_plugins()
+
+    timer = th.Timer(timeout, stop_scan, (scanner,))
+    timer.start()
+    scanner.start()
```

Comment:
The w3afCore might raise exceptions, please handle them. You might find which exceptions are raised in the console UI / GUI code.